### PR TITLE
[Finishes #187354261] buyer wish list for products

### PR DESCRIPTION
**Pull request by P-Rwirangira**
## Purpose

The purpose of this pull request is to implement the backend functionality to handle the addition of items to the buyer's wish list. This includes storing the wish list in the buyer's account and sending a confirmation message to the frontend.
## Changes Made

   -Implementing functionality to handle POST requests for adding an item to the wish list.
   -Validating that the buyer is logged in before allowing the addition of items to the wish list with the help of middlewares.
   -Adding the item to the buyer's wish list
   -Retrieving the product details (name, price, image) from the database.
   -Storing the updated wish list in the buyer's account
   -Displaying a confirmation message to the buyer

Provide instructions on how to test the changes

   -Send a POST request to the appropriate endpoint (/add-to-wishlist/productId) 
   -Verify that the item is added to the buyer's wishlist
   -Check that the product details (name, price, image) are correctly retrieved from the database
   -Ensure that the updated wishlist is stored in the buyer's account
   -Verify that a confirmation message is displayed 

## Related Issues

  - No issues

Reference any related issues or pull requests

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's co
Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's coding standards
- [x] Changes are covered by tests
- [x] Documentation is updated (if applicable)
- [x] All tests pass

### CODE REVIEW

File: src/controllers/wishlistController.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+export const addToWishlist = async (req: Request, res: Response) => {
+  const { productId } = req.body;
+  const user = req.user as UserInterface;
+
+  try {
+    const product = await Product.findOne({
+      where: { id: productId },
+    });
+    if (!product) {
+      return sendErrorResponse(res, 'Product not found');
```
Comment (hozayves) [inline src/controllers/wishlistController.ts:23]: wonderful

File: src/controllers/wishlistController.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      images: product.images,
+    });
+
+    return res.status(201).json({
+      ok: true,
+      message: 'Product added to wishlist successfully',
+      data: wishlistItem,
+    });
+  } catch (err) {
+    console.log(err);
```
Comment (hozayves) [inline src/controllers/wishlistController.ts]: Remove console.log, use logger.error instead

**Review by hozayves — CHANGES_REQUESTED** [review]
Hi Patrick,

Great work on this! I have a few suggestions. Could we consider removing the name, images, and price columns in your wishList migration? Since we can retrieve data through the product ID, these columns might be redundant.

File: src/controllers/wishlistController.ts (hunk 2 above)
Comment (P-Rwirangira) [reply src/controllers/wishlistController.ts]: I was using that when i had a bug ,but it's no longer there

File: src/controllers/wishlistController.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      include: {
+        model: Size,
+        attributes: ['productId'],
+      },
+    });
+    const productIds = wishlistItems.map((item: any) => item.Size.productId);
+    const wishlistproducts = await Product.findAll({
+      where: { id: productIds },
+    });
+    return res.status(200).json({
```
Comment (hozayves) [inline src/controllers/wishlistController.ts:68]: carbon (1)

Good work brother! I suggest this approach to consolidate crucial information into a unified response.

**Review by amin-leon — APPROVED** [review]
why haven't you tried association between user(buyer) and wishList??

**Review by hozayves — APPROVED** [review]

**Review by patrickhag — APPROVED** [review]

File: src/controllers/wishlistController.ts (hunk 3 above)
Comment (P-Rwirangira) [reply src/controllers/wishlistController.ts:68]: Done

**Review by gracemugwanezak — APPROVED** [review]

**Review by niyontwali — APPROVED** [review]

**Review by JeanIrad — APPROVED** [review]

**Pull request merged**